Cjkjvfnby/freeorion
# Inspector

### Add listener and handler feature
- allow to execute python code with out modifying any AI files just by adding entries in config.
- Move part of chart print as handler. (other part is still present in code)
### Add code to generate stub for interfaces.
- `ide_tools/` module with code related to inspector.
-  `inspect_freeOrionAIInterface.py` handler for freeOrionAIInterface
- `inspect_interface_config.ini` config for `--ai-config`. Dont forget to use game with single AI.
- `ide_tools/result/freeOrionAIInterface.py` file with stub. I add it to repo to track changes.
### How to test
- run game with `--ai-config <path to inspect_interface_config.ini >`
- start game with single AI
- After game auto exit to main menu check that stub creation date is updated (I prefer to delete it before)
### Known issues:

I don't know how to run this on universe generator without hack. Looks like we need some place for common tools, that will be added to python path for AI, universe_generator, turn_events.

Works only if AI folder has write access (I set `resource-dir` in config to my repo folder)
### TODO
- Write errors to separate log file, not to game log.
